crates/solvers/src/run.rs

```diff
@@ -0,0 +1,43 @@
+#[cfg(unix)]
+use tokio::signal::unix::{self, SignalKind};
+use {crate::domain::baseline, clap::Parser, std::net::SocketAddr, tokio::sync::oneshot};
+
+pub async fn run(args: impl Iterator<Item = String>, bind: Option<oneshot::Sender<SocketAddr>>) {
```

Comment:
bikeshed (so feel free to ignore): I would put this in `lib`. It seems like a "top-level" function (so in the `[lib.path]` just like `main` goes in the `[[bin.path]]`).

Reply:
I find it a little cleaner in it's own module. One thing I noticed in Driver, is that we were importing things for the main function in lib.rs, and then accidentally `use`ing them from other modules. For example, in lib.rs we would `use ethcontract::Address` because it is needed for `fn run()` and then in some unrelated module `foo/bar/mod.rs` we would `use crate::Address` instead of `use ethcontract::Address`. In lib.rs you can't distinguish between "import for this module" or "import for all sub modules".

Reply:
Its a fair point and I definitely agree that re-using parent module `use` statements is brittle (it can cause unrelated changes in refactors).